### PR TITLE
Backup important wiki information

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes: https://github.com/dotnet/corefx/issues/36057

Original documents in this PR:
- ~~[Building .NET Core 2.X on FreeBSD](https://github.com/dotnet/corefx/wiki/Building-.NET-Core--2.x-on-FreeBSD)~~
- [Building .NET Core 3.X on FreeBSD](https://github.com/dotnet/corefx/wiki/Building-.NET-Core-3.x-on-FreeBSD)
- ~~[Links](https://github.com/dotnet/corefx/wiki/Links)~~
- [Pick Issue](https://github.com/dotnet/corefx/wiki/Pick-issue) - Only rescued two labels mentioned here that we were not mentioning in the CoreFX documentation.
- Also added the links to the FreeBSD instructions in unix-instructions.md.

### CODE REVIEW

File: Documentation/building/freebsd-instructions-2.x.md [hunk 1]
```diff
@@ -0,0 +1,390 @@
+# FreeBSD 3.X build instructions
```
Comment (jkotas) [inline Documentation/building/freebsd-instructions-2.x.md]: The file name says 2.x, but the title says 3.x.

This should just say FreeBSD build instructions, and we should have just the latest ones. We version the build instructions together with the code. E.g. master has build instructions for master; release/2.0 has the build instructions for release/2.0; etc.

Comment (carlossanlop) [reply Documentation/building/freebsd-instructions-2.x.md]: These are the instructions to build 2.X. I had a typo. But based on your comment, I'll remove the file.

File: Documentation/project-docs/conferences.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+## .NET Conferences and Announcements links
```
Comment (jkotas) [inline Documentation/project-docs/conferences.md]: I do not see a single conference listed in this list

File: Documentation/project-docs/conferences.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    * [2.0 APIs](https://github.com/dotnet/standard/blob/master/docs/faq.md#if-there-is-no-breaking-change-why-call-it-net-standard-20)
+    * [API portability](https://github.com/Microsoft/dotnet-apiport/blob/master/README.md) and [API lookup](https://apisof.net)
+    * [Platform compat](https://github.com/dotnet/platform-compat)
+* Innovation:
+    * [Bing runs on .NET Core 2.1](https://blogs.msdn.microsoft.com/dotnet/2018/08/20/bing-com-runs-on-net-core-2-1/)
+    * [.NET Core 2.1 perf improvements](https://blogs.msdn.microsoft.com/dotnet/2018/04/18/performance-improvements-in-net-core-2-1/)
+    * [.NET Core 2.0 perf improvements](https://blogs.msdn.microsoft.com/dotnet/2017/06/07/performance-improvements-in-net-core/)
+* [.NET Standard versions](https://github.com/dotnet/standard/blob/master/docs/versions.md)
+    * Next .NET Standard plans - [.NET Standard 2.1](https://github.com/dotnet/standard/blob/master/docs/planning/netstandard-2.1/README.md#net-standard-21)
+* [CoreFX contributors](https://github.com/dotnet/corefx/graphs/contributors)
```
Comment (jkotas) [inline Documentation/project-docs/conferences.md]: Why is this here? It is neither conference or announcement.

File: Documentation/project-docs/conferences.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+* [.NET Core 3.0 previews](https://github.com/dotnet/core/tree/master/release-notes/3.0)
+    * [.NET Core 3.0 Preview 5](https://devblogs.microsoft.com/dotnet/announcing-net-core-3-0-preview-5/) (on Mon 5/5)
+    * RC (Release Candidate) - July 2019 (see [.NET 5 announcement blog](https://devblogs.microsoft.com/dotnet/introducing-net-5/))
+    * GA (General Availability) - September 2019 (see [.NET 5 announcement blog](https://devblogs.microsoft.com/dotnet/introducing-net-5/))
+* [.NET Core 3 and Support for Windows Desktop Apps](https://blogs.msdn.microsoft.com/dotnet/2018/05/07/net-core-3-and-support-for-windows-desktop-applications/) - WPF, WinForms, etc.
+
+### .NET plans
+
+* .NET Core 3.1 - LTS - November 2019 (see [.NET 5 announcement blog](https://devblogs.microsoft.com/dotnet/introducing-net-5/))
+* [.NET 5](https://devblogs.microsoft.com/dotnet/introducing-net-5/) - November 2020 GA
```
Comment (jkotas) [inline Documentation/project-docs/conferences.md]: This is redundant with roadmap on https://github.com/dotnet/core. I do not think we need to be maintaining a duplicate information here.

File: Documentation/project-docs/conferences.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
+## .NET Conferences and Announcements links
+
+### .NET Core 3.0
+
+* [.NET Core 3.0 previews](https://github.com/dotnet/core/tree/master/release-notes/3.0)
```
Comment (jkotas) [inline Documentation/project-docs/conferences.md]: Information about 3.0 previews is irrelevant now that 3.0 shipped. 

This whole page looks very questionable to me. Is it really useful?

File: Documentation/building/freebsd-instructions.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
+# FreeBSD build instructions
+
+## Disclaimer
+Instructions bellow may or may not give you what you want.
```
Comment (danmosemsft) [inline Documentation/building/freebsd-instructions.md]: typo

File: Documentation/project-docs/conferences.md (hunk 4 above)
Comment (danmosemsft) [reply Documentation/project-docs/conferences.md]: I agree, I think if we need such a document as this it should probably be in dotnet/core. I'd dump it.

**Review by danmoseley — APPROVED** [review]

File: Documentation/building/freebsd-instructions.md (hunk 6 above)
Comment (stephentoub) [reply Documentation/building/freebsd-instructions.md]: In addition to the "bellow" typo, this is also a strange disclaimer.  Why have instructions if they "may or may not give you what you want?"  We should be clear about why they may not provide the desired results.

File: Documentation/building/freebsd-instructions.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+Tested on plain FreeBSD 11.3 Azure image
+
+## Prerequisites
+This needs to be done once on fresh system.
+
+```sudo pkg install cmake git icu libunwind bash python2 krb5 lttng-ust llvm60 libgit2```
+
+some scripts may still assume /bin/bash exists. To workaround it for now do of needed:
```
Comment (stephentoub) [inline Documentation/building/freebsd-instructions.md]: "now do of needed"?

File: Documentation/building/freebsd-instructions.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+```
+sudo ln -s /usr/local/bin/bash /bin/bash
+```
+
+This is certainly undesirable and it should be avoided if possible.
```
Comment (stephentoub) [inline Documentation/building/freebsd-instructions.md]: What is undesirable?

File: Documentation/building/freebsd-instructions.md [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+
+# Get bootstrap cli
+```
+mkdir ~/dotnet
+cd ~/dotnet
+curl https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-freebsd-x64.tar.gz | tar xfz - 
+```
+if on 12.x you may also need to set `LD_PRELOAD` to `/usr/lib/libpthread.so` to avoid issue when cli freezes.
```
Comment (stephentoub) [inline Documentation/building/freebsd-instructions.md]: "to avoid issue" => "to avoid issues"

And what does it mean that "cli freezes"?

File: Documentation/building/freebsd-instructions.md [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+# Get bootstrap cli
+```
+mkdir ~/dotnet
+cd ~/dotnet
+curl https://dotnetcli.blob.core.windows.net/dotnet/Sdk/master/dotnet-sdk-latest-freebsd-x64.tar.gz | tar xfz - 
+```
+if on 12.x you may also need to set `LD_PRELOAD` to `/usr/lib/libpthread.so` to avoid issue when cli freezes. 
+
+
+As of summer 2019 this CLI is no longer good enough to build all repos. If that is your case jump to section [Updating CLI](#updating--bootstrap-cli)
```
Comment (stephentoub) [inline Documentation/building/freebsd-instructions.md]: "If that is your case"... what case?  It's beyond that point for everyone, no?

File: Documentation/building/freebsd-instructions.md [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+Binary snapshot can be obtained from https://github.com/wfurt/blob as dotnet-sdk-freebsd-x64-latest.tgz
+
+## Getting sources
+master of source-build pulls in source code of specific snapshot instead of tip of master branches.
```
Comment (stephentoub) [inline Documentation/building/freebsd-instructions.md]: I'm having a hard time following these docs.  I'm going to stop commenting on it.

File: Documentation/building/unix-instructions.md [hunk 12]
```diff
@@ -1,5 +1,8 @@
 Building CoreFX on FreeBSD, Linux and OS X
 ==========================================
+
+Build instructions for FreeBSD can be found [here](freebsd-instructions.md)
```
Comment (stephentoub) [inline Documentation/building/unix-instructions.md]: FreeBSD isn't supported, whereas Linux and macOS are.  Seems like "Building CoreFX on FreeBSD, Linux, and OSX" should instead be "Building CoreFX on Linux and macOS", and then this line should be in its own section later.

File: Documentation/project-docs/conferences.md (hunk 5 above)
Comment (stephentoub) [reply Documentation/project-docs/conferences.md]: Information on this page is the purpose of the dotnet/core repo.  I would just delete this whole .md file.

**Comment by carlossanlop** [comment]
Thanks for your comments, @stephentoub @jkotas @danmosemsft .

- I removed the conferences file. It seems it only contains links to announcements which, as Stephen mentioned, we already post regularly on dotnet/core.
- I removed "FreeBSD" from the title in `unix-instructions.md`.
- I still think we should bring the FreeBSD 3.X instructions to corefx, since the document has been modified recently multiple times (see the [commit history](https://github.com/dotnet/corefx/wiki/Building-.NET-Core-3.x-on-FreeBSD/_history)). The main problem is the language used to write the document, so we could ask the people involved in this document in the past to help reword the document, or I can do it myself, but I would do it in a future PR. What are your thoughts?

File: Documentation/building/freebsd-instructions.md (hunk 11 above)
Comment (carlossanlop) [reply Documentation/building/freebsd-instructions.md]: Yes I see what you mean. The document needs to be rewritten.

**Comment by stephentoub** [comment]
Sounds fine. Thanks.

File: Documentation/building/unix-instructions.md (hunk 12 above)
Comment (carlossanlop) [reply Documentation/building/unix-instructions.md]: I'll move it to its own section at the end of this document.
